routes/dmthreads.js

```diff
@@ -0,0 +1,305 @@
+/****************************************************************************
+ * TO DO IN THIS FILE: 
+ * Add dmEnabled Check (sendoer + recipient)
+ * Consider updating the route to be more REST-like
+ ****************************************************************************/
+import {Router} from "express";
+import mongoCollections from "../config/mongoCollections.js";
+import {
+    createThread,
+    createMessage,
+    addMessageToThread,
+    fetchThreadById,
+    fetchThreadsForUser
+} from "../data/dmthreads.js";  
+
+const {dmthreads} = mongoCollections;
+const router = Router();
+const lastMessageTimestamps = {}; // In Mem Rate Limiter, mapped to username -> timestamp of last message sent.
+const {users: usersCollection} = mongoCollections; // Only for accessing Banlist Table.
+
+/****************************************************************************
+ * Function getUserByUsername(username)
+ * Helper Function to access Ban Status
+ ****************************************************************************/
+async function getUserByUsername(username) {
+    const users = await usersCollection();
+    return await users.findOne({ "profile.username": username.toLowerCase() });
+}
+
+/****************************************************************************
+ * GET .../dmthreads/create
+ * Show the Create Thread Form
+ ****************************************************************************/
+router.get("/create", async (req, res) => {
+    if (req.session.user.isBanned) {
+        return res.status(403).render("error", {
+            error: "You cannot use DMs while banned."
+        });
+    }
```

Comment:
I think a simple middleware like 
```suggestion
const requireNotBanned = (req, res, next) => {
    if (req.session.user?.isBanned) {
        return res.status(403).render("error", {
            error: "You cannot use direct messages while banned."
        });
    }
    next();
};
```

Maybe we can add a validation function as well for users who are not the logged in user. 
This can help reduce a lot of code repetition